core/storage/statistics/gae_models.py

```diff
@@ -82,6 +82,10 @@ class AnswerSubmittedEventLogEntryModel(base_models.BaseModel):
     time_spent_in_state_secs = ndb.FloatProperty()
     # Whether the submitted answer received useful feedback
     is_feedback_useful = ndb.BooleanProperty(indexed=True)
+    # The version of the event schema used to describe an event of this type.
+    # Details on the schema are given in the docstring for this class.
+    event_schema_version = ndb.IntegerProperty(
+        indexed=True, default=feconf.CURRENT_EVENT_MODELS_SCHEMA_VERSION)
```

Comment:
I don't think you can specify a default like this; existing events will then default to the current schema version. Instead, I suggest adding def put() and having that check for a non-None event schema version on newly-put models before passing it to the superclass put(). Ditto below.

Or alternatively the put() could do "self.event_schema_version = feconf.CURRENT_EVENT_MODELS_SCHEMA_VERSION)" prior to delegating to the superclass method. That way, clients don't have to care about passing it in at all, and it's automatically handled correctly.

(This works because it will affect only newly-put models and not existing ones.)

Ditto elsewhere.

Reply:
@seanlip. I did not add a new default field by the way. For all existing event models, this field already existed. I just changed it from a class specific variable to a variable specified in feconf. So, if this field already exists for all current event models and no future models have been recorded yet, why not keep this?

Reply:
Oh, nice! Yes, that's fine then. Ignore my comment (and thanks for clarifying).